### PR TITLE
feat: UI element to modify tradition metadata (#97)

### PR DESCRIPTION
**Pull request by jorisvanzundert**
Implementes #97.

### CODE REVIEW

**Review by peter-gy — COMMENTED** [review]
Great additions to the UI. What do you think about my suggestions?

File: frontend/www/src/js/modules/dashboard/tradition/state.js [hunk 1]
```diff
@@ -51,6 +51,26 @@ class TraditionStore extends StateStore {
...
+  updateTradition( tradition ) {
+    const tradIdx = this.state.availableTraditions.findIndex( availableTradition => availableTradition.id == tradition.id );
+    if( tradIdx > -1 ) {
+      let availableTradition = this.state.availableTraditions[ tradIdx ];
+      availableTradition = Object.assign( availableTradition, tradition );
+      this.setState({
+        ...this.state
+      });
+    }
+  }
```
Comment (peter-gy) [inline frontend/www/src/js/modules/dashboard/tradition/state.js:72]: I would avoid updating by reference as it usually leads to very hard-to-debug errors.
What about this version?
```suggestion
  updateTradition(tradition) {
    const tradIdx = this.state.availableTraditions.findIndex(availableTradition => availableTradition.id == tradition.id);
    const traditionFound = tradIdx > -1;
    if (traditionFound) {
      const traditionToUpdate = this.state.availableTraditions[tradIdx];
      const updatedTradition = {...traditionToUpdate, ...tradition};
      this.state.availableTraditions[tradIdx] = updatedTradition;
      this.setState({...this.state});
    }
  }
```

File: frontend/www/src/js/modules/common/state/store.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ class StateStore {
    */
   setState(newState) {
     // Only execute this operation if the state has actually changed.
-    const shouldSkip = objectsEqual(this._state, newState);
+    const shouldSkip = Object.is(this._state, newState);
```
[the remark this answers is not in the document]
Comment (jorisvanzundert) [reply frontend/www/src/js/modules/common/state/store.js:33]: Ugh.. this is where JS is really hardheaded. This is the problem in shorthand:

```
const o1 = { a: { b: "c" } }
const o2 = o1
console.log( objectsEqual( o1, o2 ) ) //true
o2.a.b = "d"  // { a: { b: "d" } }
console.log( objectsEqual( o1, o2 ) ) //true 
```

The problem is that JS only does a shallow check. This means your suggestion doesn't solve it. But no worries, I have a variation on your solution that works.

File: frontend/www/src/js/modules/common/components/dialog.js [hunk 3]
```diff
@@ -92,8 +92,16 @@ class StemmawebDialog extends HTMLElement {
...
+      const handlerResult = ( actions.onOk || (() => {}) )();
+      // Differentiate between handlers that are just handlers, and
+      // handlers that are type Promise
+      if( typeof handlerResult.then === 'function' ){
+        handlerResult.then( (promise) => { 
+          if( promise.success ) {
+            dialogInstance.hide();
+          }
+        } )
+      }
```
[the remark this answers is not in the document]
Comment (jorisvanzundert) [reply frontend/www/src/js/modules/common/components/dialog.js:104]: :)

**Comment by jorisvanzundert** [comment]
I have added a few updates to this PR in response to requests by @peter-gy. 
Also started working no section functionality (display/change), ongoing work.
Changing **Tradition** metadata should now be fully functional.

Tests to be added:

* Update/change of 
    * name, 
    * access, 
    * language, 
    * and direction 
using the edit button (top right hand side) should be reflected in the properties shown in the table.
* If the name changes, that should also be visible in the main page (top of page has the name/title of the Tradition).

Note that this PR also starts implementing displaying sections and display/change of section metadata, but this is not test ready yet.

**Comment by jorisvanzundert** [comment]
This PR has been superseded by https://github.com/tla/stemmaweb/pull/154 and contains at most a lot of stuff that is now very stale. Won't merge, closing this one.

**Pull request closed without merging**